### ISSUE TITLE
Redirect for Records > Reconstructing Records based on rewriting work

### ISSUE THREAD

**Issue by mnorthuis** [department-of-veterans-affairs/va.gov-team#952]
## User Story
As a user, if I try to access https://www.benefits.va.gov/COMPENSATION/NPRC1973Fire.asp, I am immediately redirected to the new version of this page at /records/get-military-service-records/reconstruct-records/.

*Original issue in Vets repo: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18622
Note - this ticket is for redirecting only one of the URLs referenced in the original ticket because they are very different content. A separate issue will be opened for the other reference legacy page.*

## Goal
Users who navigate or link to the legacy version of the page are immediately redirected to the new page.  

## Acceptance Criteria
- [x] Determine if all parity changes are complete/live and log tickets that track the approval status
- [x] Validate previously documented redirects and update as necessary
- [x] Create a new parity redirect issue in va.gov-team repo with the list of redirects and reference the original issue and the content approval ticket
- [x] Close the old issues in vets.gov-team repo and reference the new issue in va.gov-team repo
- [x] Communicate upcoming redirects to stakeholders prior to implementing
- [x] Assign out tickets to front-end or devops as appropriate to implement redirects
- [x] Validate redirects in prod and update documentation when complete

## Definition of Done
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `call center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

Current URL  | Redirect to
-- | --
https://www.benefits.va.gov/COMPENSATION/NPRC1973Fire.asp | /records/get-military-service-records/reconstruct-records/

**Comment by mnorthuis** [comment]
Rewriting work was approved and implemented and new page is live per this issue: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18470

**Comment by mnorthuis** [comment]
Blocked until ops/front end team can determine approach for implementation of redirects.

**Comment by Karak888** [comment]
This page is live- redirect work needs FE engineering.

**Comment by mnorthuis** [comment]
@jenniferlee-dsva who is the stakeholder responsible for this legacy content?  They should be notified prior to us redirecting the old URL.

**Comment by jenniferlee-dsva** [comment]
I am not sure, bc Records as a benefit category doesn't exist in VA org world. I think we can share it with Misty Sweet though and ask her to route it to the right SME. 

Do you mind sending that email to Misty and team (Sandy Tadeo, Adrianne Hooten) and cc'ing me?

**Comment by mnorthuis** [comment]
@Karak888 @jenniferlee-dsva Contacts for the review of this content were Misty, Elana, Sandy and Adrianne.  Per this issue https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18470, that is who approved the page.

**Comment by mnorthuis** [comment]
Blocked until redirect approach is determined.

**Comment by mnorthuis** [comment]
@Karak888 This redirect can be implemented once the appropriate stakeholder is notified.

**Comment by Karak888** [comment]
@mnorthuis just checking to see if you contacted the SMEs you referenced above as contacts for the review of this content (Misty, Elana, Sandy and Adrianne) per the old vets.gov ticket #18470 or does someone need to reach out still to SME before we can implement redirects?  

@jenniferlee-dsva @AnneHurley can one of you please contact the correct SME so we can get this redirect taken care of today if possible?

**Comment by Karak888** [comment]
@AnneHurley @jenniferlee-dsva - per Mikki it is on our team to reach out to the SMEs to get approval- Can someone please do this so that we can hopefully get this one closed out?  Thanks!!

**Comment by Karak888** [comment]
ready to redirect old page to new page - when can we have this be executed so @jenniferlee-dsva can communicate that to stakholder (timeframe)- 

we would have to ask stakeholder to update old page to replace this redirect

@ncksllvn @jenniferlee-dsva to talk to get an estimated timeframe at last a weeks notice for communication to stakeholders.

**Comment by ncksllvn** [comment]
Because requests to benefits.va.gov don't pass through our servers (we only receive traffic for www.va.gov), we can't do a server-side redirect. We would need to reach out to partners and do some coordination to figure out how to do it server-side. 

However, we have the immediate ability to implement this redirect client-side via JavaScript, and it would be easy. This is because we have some of our JavaScript code being executed on benefits.va.gov in order to render the WBC header/footer. We're actually already redirecting some pages, like http://www.benefits.va.gov/compensation/types-disability.asp.

The downside of doing it client-side though is that the redirect is not performant (notice the page is already loading up once the redirect executes) and we don't get the same SEO benefits as a server-side redirect. But again, we have no blockers to do it client-side, and it's very easy technically.

**Comment by Karak888** [comment]
Thanks @ncksllvn  I will let @jenniferlee-dsva make final decision, but if it is for this one page and it is that easy we should probably go ahead and do it while we decide collectively with VSP how we will handle the TeamSite redirects.  I am guessing we can start to think about having a conversation for these one-offs with insights team perhaps?  Hopefully this will be our last one-off :) THANKS for the info Nick!!! @jenniferlee-dsva I'll let ya make the decision here!

**Comment by mnorthuis** [comment]
@Karak888 My documentation is updated and this redirect can be assigned to your front end team for a client side redirect when your team is ready for implementation.

**Comment by jenniferlee-dsva** [comment]
@Karak888 @ncksllvn - let's use this as our test 'archive TeamSite page' and redirect using our new approach. I will send out meeting invite and invite you both. The redirect part is obvs not in question; the convo will be to work with Sandy Tadeo to test archiving this page in TeamSite and seeing if the redirect still works via header/footer code. 

FYSA @mnorthuis @meganhkelley @batemapf

**Comment by Karak888** [comment]
@lunascoop I will take a look.  Just updated it above to say this was separate from the Benefit Hub URLs for redirect.  Can you please confirm they are ineed 2 separate tasks for you? Thanks!

**Comment by jenniferlee-dsva** [comment]
@Karak888 @ncksllvn @lunascoop - please size and assess when redirect can be put in queue and communicated. Let's use our new process!

**Comment by aeltarhoni** [comment]
@lunascoop will meet on Monday with Nick and Kara to discuss next steps.

**Comment by ncksllvn** [comment]
I don't see a reason why we can't implement this immediately like our other client-side redirects. @jenniferlee-dsva was the client-side redirect was considered too slow?

**Comment by lunascoop** [comment]
@Karak888 @ncksllvn @jenniferlee-dsva I will send out the redirect notice email to stakeholders with **12/19** as the effective date.

**Comment by Karak888** [comment]
@ncksllvn @kelsonic - this is for redirect of records that @mnorthuis said she was fine on her end. 
**Redirect Request**:  
- **Current URL**:  https://www.benefits.va.gov/COMPENSATION/NPRC1973Fire.asp
- **Redirect to**:  /records/get-military-service-records/reconstruct-records/

**Note:**  Client communications ^ were sent out for 12/19/19 

CC:  @jenniferlee-dsva

**Comment by ncksllvn** [comment]
Thanks @Karak888, I'm opening a PR now

**Comment by ncksllvn** [comment]
Here's the PR for reference, https://github.com/department-of-veterans-affairs/vets-website/pull/11311.

**Comment by ncksllvn** [comment]
I don't think we should close this. My PR won't be merged until tomorrow.

**Comment by lunascoop** [comment]
> I don't think we should close this. My PR won't be merged until tomorrow.

Agreed @ncksllvn. I will close the ticket after I validate the redirect per our process.

**Comment by ncksllvn** [comment]
My PR implementing the redirect is merged and should be on the site in less than an hour

**Comment by lunascoop** [comment]
@ncksllvn @mnorthuis @Karak888 @jenniferlee-dsva I've validated that this redirect is live. I will close this ticket and send notification to stakeholders.

**Comment by ncksllvn** [comment]
Woohoo! I actually merged this right at 1:59 PM ET. Was a close one 😅

**Comment by jenniferlee-dsva** [comment]
Living on the edge @ncksllvn

**Comment by Karak888** [comment]
Well done!!!
